lib/assets/chart_builder/main.js

```diff
@@ -0,0 +1,223 @@
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+          <div class="group">
+            <div class="field">
+              <label class="input-label">Y Axis</label>
+              <select class="input input--sm" name="y_axis">
+              </select>
+            </div>
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="y_axis_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+          <div class="group">
+            <div class="field">
+              <label class="input-label">Color</label>
+              <select class="input input--sm" name="color">
+              </select>
+            </div>
+            <div class="field">
+              <label class="input-label">Type</label>
+              <select class="input input--sm" name="color_type">
+                <option value=""></option>
+                <option value="quantitative">Quantitative</option>
+                <option value="nominal">Nominal</option>
+                <option value="ordinal">Ordinal</option>
+              </select>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  `;
+
+  updateInfoBox(info.missing_dep);
+  setDataOptions(info.options);
+  setValues(info.fields);
+
+  const hasType = ["x_axis", "y_axis", "color"];
+  const types = ["x_axis_type", "y_axis_type", "color_type"];
+
+  ctx.root.addEventListener("blur", handleFieldChange, true);
+  ctx.root.addEventListener("change", handleFieldChange);
+
+  function handleFieldChange(event) {
+    const { name, value } = event.target;
+    if (hasType.includes(name)) { maybeReset({name, value}) }
+    ctx.pushEvent("update_field", { field: name, value });
+  }
+
+  ctx.handleEvent("update", ({ fields }) => {
+    setValues(fields);
+  });
+
+  ctx.handleEvent("missing_dep", ({ dep }) => {
+    updateInfoBox(dep);
+  });
+
+  ctx.handleEvent("set_axis_options", ({ options }) => {
+    options ? setAxisOptions(options) : missingData();
+  });
+
+  ctx.handleEvent("set_data_options", ({ options }) => {
+    setDataOptions(options);
+  });
+
+  function setValues(fields) {
+    for (const field in fields) {
+      ctx.root.querySelector(`[name="${field}"]`).value = fields[field];
+    }
+  }
+
+  function setDataOptions(options) {
+    const selectElem = ctx.root.querySelector(`[name="data"]`);
+    selectElem.innerHTML = "";
+    createOptions(selectElem, Object.keys(options))
+    const selected = selectElem.value;
```

Comment:
There's a catch! Let's say we have a variable with data and the chart builder below using this variable. Now when we re-open the notebook (or restart the runtime and install the deps), the smart cell would start, but it would receive an empty binding, until the above code is executed. In such case, we want to preserve the previously selected variable name (that we persist via `attrs`) until the user changes it, regardless if it's currently available. This is important when there are more options, because we don't want to change it implicitly. It's analogous to the connection variable in #105, here's an example:

https://user-images.githubusercontent.com/17034772/158814743-f5fe9837-b701-4900-b899-5627484f8271.mp4

We can handle it the same way, in particular see the `function renderConnectionSelect` function.